src/components/SuggestionsList/SuggestionsListItem.tsx

```diff
@@ -5,13 +5,17 @@ import { NestableBaseComponentProps } from '../../util/BaseComponent/props';
 import MediaObject, { MediaObjectSize } from '../MediaObject';
 import Avatar, { AvatarSize } from '../Avatar';
 import Text from '../Text';
+import BadgeFlexternal from '../Icon/icons/BadgeFlexternal';
+import { IconSize } from '../Icon';
+
 import './SuggestionsListItem.css';
 
 export interface SuggestionItem {
   id: string | number;
   imageUrl?: string;
   name: string;
   description?: string;
+  isFlexternal: boolean;
```

Comment:
i really don't like that this would bake Yammer business logic directly into SuggestionsList, YamUI should be enforcing the visuals/layout but not the actual content like "BadgeFlexternal".

We're also not yet looking at this holistically because we haven't implemented the other SuggestionsListItem templates `text` | `mediaRound` | `mediaSoft` | `mediaSquare`: https://github.com/Microsoft/YamUI/issues/167

i think i would prefer SuggestionsListItem to accept badgeContent or AvatarProps, somehow exposing a way for the client to set badgeContent without dictating what that content should actually be. it seems that only `mediaRound` will optionally allow badge content via Avatar, so this specific item interface could also accept AvatarProps?

Reply:
`isFlexternal` is a Yammer business concern, not a presentational concern

Reply:
additionally we'd have to keep on adding new props for every new badge type we want to manually support here...

Reply:
hmm - calculating isFlexternal is Yammer business logic, but supporting a badge type called Flexternal isnt particularly. It's just a dev convenience. Could change this type a badgeType?: 'globe' | 'star' | 'face' if that helps?

Reply:
badge content is not the concern of SuggestionsList, it's really the concern of Avatar. even adding badgeType prevents SuggestionsList from being used in different ways. what if we use SuggestionsList to search for attachments, and we want badgeContent to be the file type icon? what if we want to show people's presence as the badge content? i don't think SuggestionsList should be gatekeeper, it's displaying an Avatar so i think Avatar should be the content gatekeeper.

i totally agree that "Flexternal" is a bad name for an icon, maybe we should separately rename that to something generic like "globe"

Reply:
I'm fine with us avoiding Yammer domain terms like Flexternal. It seems like `SuggestionItem` should extend `AvatarProps`. Agree?

Reply:
i think it would be better for SuggestionItem to accept an avatarProps field, and SuggestionItem could splat them into its Avatar before setting its own props over the top

Reply:
ok